picard/script/functions.py

```diff
@@ -1517,3 +1533,72 @@ def func_cleanmulti(parser, multi):
     values = [str(value) for value in parser.context.getall(name) if value or value == 0]
     parser.context[multi] = values
     return ""
+
+
+def _type_args(_type, *args):
+    haystack = set()
+    # Automatically expand multi-value arguments
+    for item in args:
+        haystack = haystack.union(set(x for x in item.split('; ')))
+    _typer = None
+    if _type == 'int':
+        _typer = int
+    elif _type == 'float':
+        _typer = float
+    elif _type in ('text', 'ncase'):
+        pass
+    else:
+        # Unknown processing type
+        raise ValueError
+    if _typer is not None:
+        haystack = set(_typer(item) for item in haystack)
+    return haystack
+
+
+def _extract(_func, _type, *args):
+    try:
+        haystack = _type_args(_type, *args)
+    except ValueError:
+        return ""
+
+    if _type == 'ncase':
```

Comment:
What does `ncase` mean?

Reply:
It's a non case-sensitive text comparison.  I'm open to suggestions for a less cryptic / confusing name.  :wink: